src/drivers/imx/edma.c

```diff
@@ -325,10 +325,10 @@ static int edma_setup_tcd(struct dma_chan_data *channel, int16_t soff,
 		return rc;
 
 #ifdef CONFIG_IMX8ULP
-	if (channel->direction == DMA_DIR_MEM_TO_DEV)
-		dma_chan_reg_write(channel, EDMA_CH_MUX, IMX8ULP_DMAMUX2_SAI5_TX);
-	else
-		dma_chan_reg_write(channel, EDMA_CH_MUX, IMX8ULP_DMAMUX2_SAI5_RX);
+	if (!dma_chan_reg_read(channel, EDMA_CH_MUX))
+		(channel->direction == DMA_DIR_MEM_TO_DEV) ?
+			dma_chan_reg_write(channel, EDMA_CH_MUX, IMX8ULP_DMAMUX2_SAI5_TX) :
+			dma_chan_reg_write(channel, EDMA_CH_MUX, IMX8ULP_DMAMUX2_SAI5_RX);
```

Comment:
You added `:` instead of `; I'm not sure this code compile. 

Which reminds me we should add imx8ulp compilation in CI.

Reply:
He added a ternary operator, so this should compile fine.